### PR TITLE
Add error codes for sub organization tenant deletion

### PR DESCRIPTION
**Pull request by RushanNanayakkara**

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 1]
```diff
@@ -540,7 +540,23 @@ public enum ErrorMessages {
...
                         " parent application: %s in organization: %s."),
         ERROR_CODE_ERROR_DELETING_USER_ROLE_ASSIGNMENTS("65098", "Unable to remove the user from the roles.",
-                "Server encountered an error while removing the user id: %s from the roles.");
+                "Server encountered an error while removing the user id: %s from the roles."),
+        ERROR_CODE_ERROR_AUTHENTICATING_TENANT_DELETION_USER("65099", "Error while authenticating" +
+                " tenant deletion user",
+                "Error while sending request for generating authentication token for tenant deletion user"),
+        ERROR_CODE_ERROR_BUILDING_TENANT_DELETION_AUTH_REQUEST_BODY("65100", "Error while authenticating" +
+                " tenant deletion user",
+                "Error while sending request for generating authentication token for tenant deletion user"),
```
Comment (piyarathnalakmali) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:549]: Both has the same message and description.

Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: fixed with commit

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 2]
```diff
@@ -540,7 +540,22 @@ public enum ErrorMessages {
                 "Server encountered an error when deleting shared application roles related to" +
                         " parent application: %s in organization: %s."),
         ERROR_CODE_ERROR_DELETING_USER_ROLE_ASSIGNMENTS("65098", "Unable to remove the user from the roles.",
-                "Server encountered an error while removing the user id: %s from the roles.");
+                "Server encountered an error while removing the user id: %s from the roles."),
+        ERROR_CODE_ERROR_AUTHENTICATING_TENANT_DELETION_USER("65099", "Error while authenticating" +
+                " tenant deletion user.",
+                "Error while sending request for generating authentication token for tenant deletion user."),
+        ERROR_CODE_ERROR_BUILDING_TENANT_DELETION_AUTH_REQUEST_BODY("65100", "Error building auth request.", "Error while building tenant deletion authentication" +
```
Comment (piyarathnalakmali) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:547]: Formatting issue. Line is longer than 120 characters

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 3]
```diff
@@ -540,7 +540,22 @@ public enum ErrorMessages {
...
+                " request body."),
+        ERROR_CODE_ERROR_DELETING_SUB_ORGANIZATION_TENANT("65101", "Unable to delete the tenant of the " +
+                "sub organization.", "Server encountered an error while sending delete request to the " +
+                "tenant deletion micro service."),
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:551]:
```suggestion
                "tenant deletion service."),
```

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 4]
```diff
@@ -540,7 +540,22 @@ public enum ErrorMessages {
...
+        ERROR_CODE_ERROR_WHILE_DELETING_TENANT_META_DATA("65102", "Unable to delete tenant meta data.",
+                "Encountered an error while attempting to delete tenant meta data."),
+        ERROR_CODE_ERROR_RETRIEVING_TENANT_DELETION_MICRO_SERVICE_HOST_URL("650103", "Tenant deletion micro " +
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:554]:
```suggestion
        ERROR_CODE_ERROR_RETRIEVING_TENANT_DELETION_MICRO_SERVICE_HOST_URL("650103", "Tenant deletion  " +
```

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 5]
```diff
@@ -540,7 +540,22 @@ public enum ErrorMessages {
...
+                "service host url not found.", "Configuration could not be found for tenant deletion micro " +
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:555]:
```suggestion
                "service host url not found.", "Configuration could not be found for tenant deletion  " +
```

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 2 above)
Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: fixed with commit

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 3 above)
Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: fixed with commit

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 4 above)
Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: fixed with commit

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java (hunk 5 above)
Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java]: fixed with commit

**Review by AnuradhaSK — APPROVED** [review]

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java [hunk 6]
```diff
@@ -540,7 +540,22 @@ public enum ErrorMessages {
...
+        ERROR_CODE_ERROR_DELETING_SUB_ORGANIZATION_TENANT("65101", "Unable to delete the tenant of the " +
+                "sub organization.", "Server encountered an error while sending delete request to the " +
+                "tenant deletion service."),
+        ERROR_CODE_ERROR_WHILE_DELETING_TENANT_META_DATA("65102", "Unable to delete tenant meta data.",
+                "Encountered an error while attempting to delete tenant meta data."),
+        ERROR_CODE_ERROR_RETRIEVING_TENANT_DELETION_SERVICE_HOST_URL("650103", "Tenant deletion " +
+                "service host url not found.", "Configuration could not be found for tenant deletion " +
+                "service host url."),
+        ERROR_CODE_ERROR_RETRIEVING_IDENTITY_SERVER_HOST_URL("65104", "Identity server host url not found.",
+                "Configuration could not be found for identity server host url."),;
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:558]: Hope the last comma character does not cause any issue.

Comment (RushanNanayakkara) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java:558]: tested and did not cause any errors.

**Pull request merged**